src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -258,12 +250,25 @@ protected override async Task WriteCacheBytesAsync(string cacheKey, byte[] bytes
                 Logger.MemoryCacheCount(_logger, _memoryCacheType, write, _memoryCache.Count, null);
             }
 
+            if ((cacheExpiry != null && _distributedCacheOptions.AbsoluteExpiration != null && _distributedCacheOptions.AbsoluteExpiration < cacheExpiry)
+               || (cacheExpiry == null && _distributedCacheOptions.AbsoluteExpiration != null))
+            {
+                cacheExpiry = _distributedCacheOptions.AbsoluteExpiration;
+            }
+
+            DistributedCacheEntryOptions distributedCacheEntryOptions = new DistributedCacheEntryOptions()
+            {
+                AbsoluteExpiration = cacheExpiry,
```

Comment:
Do you know which one of these expirations will win if they are all set? Is it the one closer to DateTime.Now ?

Reply:
Same comment for the memory cache.